binding/java/build.gradle

```diff
@@ -83,6 +86,10 @@ dependencies {
     testImplementation 'org.junit.jupiter:junit-jupiter:5.4.2'
 }
 
+compileTestJava {
```

Comment:
Could you explain "assemble" -> "jar" and this "compileTestJava" task to me?

Reply:
The new version of Gradle throws a LOT of errors with our old `build.gradle` file because it detects that certain jobs rely on these copied lib files but don't explicitly or even implicitly have them as build dependencies.

The `assemble` -> `jar` change is that `jar` is the underlying build step for `assemble` and the `jar` step is actually what requires the files to be there.

The `compileTestJava` addition is to add the files as a build dependency implicitly (via the `assembly` step, which I believe is required first anyways?) when building the tests.

Reply:
cool, thanks